src/sentry/integrations/bitbucket_server/blame.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from collections.abc import Mapping, Sequence
+from dataclasses import asdict
+from datetime import datetime, timezone
+from typing import Any
+
+from sentry.integrations.bitbucket_server.utils import BitbucketServerAPIPath
+from sentry.integrations.source_code_management.commit_context import (
+    CommitInfo,
+    FileBlameInfo,
+    SourceLineInfo,
+)
+from sentry.shared_integrations.client.base import BaseApiClient
+from sentry.shared_integrations.exceptions import ApiError
+
+logger = logging.getLogger("sentry.integrations.bitbucket_server")
+
+
+def _blame_file(
+    client: BaseApiClient, file: SourceLineInfo, extra: Mapping[str, Any]
+) -> FileBlameInfo | None:
+    if file.lineno is None:
+        logger.warning("blame_file.no_lineno", extra=extra)
+        return None
+
+    project = file.repo.config["project"]
+    repo = file.repo.config["repo"]
+
+    browse_url = BitbucketServerAPIPath.get_browse(
+        project=project,
+        repo=repo,
+        path=file.path,
+        sha=file.ref,
+        blame=True,
+        no_content=True,
+    )
+
+    try:
+        data = client.get(browse_url)
+    except ApiError as e:
+        if e.code in (401, 403, 404):
+            logger.warning(
+                "blame_file.browse.api_error",
+                extra={
+                    **extra,
+                    "code": e.code,
+                    "error_message": e.text,
+                },
+            )
+            return None
+        raise
+
+    for entry in data:
+        start = entry["lineNumber"]
+        span = entry["spannedLines"]
```

Comment:
is there an example of what the blame response looks like? i couldn't find any on the Bitbucket Server API docs